updates/v2.1.0/migrate_message_code.php

```diff
@@ -11,9 +12,28 @@ class MigrateMessageCode extends Migration
 
     public function up()
     {
+        if (!Schema::hasColumn(self::TABLE_NAME, 'code_pre_2_1_0')) {
+            Schema::table(self::TABLE_NAME, function($table) {
+                $table->string('code_pre_2_1_0')->index()->nullable();
+            });
+        }
+
         foreach (Message::all() as $message) {
-            $default_message = $message->message_data['x'];
-            $message->code = Message::makeMessageCode($default_message);
+            $message->code_pre_2_1_0 = $message->code;
+            $message->code = $message->code_md5 ?: Message::makeMessageCode($message->message_data[Message::DEFAULT_LOCALE]);
```

Comment:
Where does this `$message->code_md5` property comes from @matteotrubini ?

Reply:
`code_md5` col is created on `down()` and populated from `code` before it's recalculated with the old method.
This follow the same logic of `code_pre_2_1_0`: preserve old values on migrations to permit checks and debug.